cli/src/commands/next.rs

```diff
@@ -60,6 +62,42 @@ pub(crate) struct NextArgs {
     edit: bool,
 }
 
+pub fn choose_commit<'a>(
+    ui: &mut Ui,
+    cmd: &str,
+    commits: &'a [Commit],
+) -> Result<&'a Commit, CommandError> {
+    writeln!(ui.stdout(), "ambiguous {cmd} commit, choose one to target:")?;
+    for (i, commit) in commits.iter().enumerate() {
+        writeln!(
+            ui.stdout(),
+            "{}: [{}] {}",
+            i + 1,
+            short_commit_hash(commit.id()),
+            commit.description().lines().next().unwrap_or_default()
+        )?;
+    }
+    writeln!(ui.stdout(), "q: quit the prompt")?;
+
+    let mut choices: Vec<String> = Default::default();
+    for i in 1..(commits.len() + 1) {
+        choices.push(format!("{i}"));
+    }
+    choices.push("q".to_string());
+    let refs: Vec<&str> = choices.iter().map(|s| s.deref()).collect();
+
+    let choice = ui.prompt_choice(
+        "enter the index of the commit you want to target",
+        &refs,
+        None,
+    )?;
+    if choice == "q" {
+        return Err(user_error("ambiguous target commit"));
+    }
+
+    Ok(&commits[choice.parse::<usize>().unwrap() - 1])
+}
```

Comment:
Not blocking, but it could be valuable to move this function to something like `movement_utils.rs` as both `jj edit @+/@-` (the old way of `next/prev`) also runs into this issue.

Reply:
I thought of that too (moving this to a helper) but without a coded 3rd use case I'm not sure what it should look like / what that helper file should be named.  Edit currently calls `resolve_single_rev()` so would need more (likely controversial) work to make use of this.

Reply:
For me it's fine to defer this to later so if you also want to tackle the `jj edit` problem, we can decide then.  

> I thought of that too (moving this to a helper) but without a coded 3rd use case I'm not sure what it should look like / what that helper file should be named.

That's fair.